x-pack/plugins/alerting/server/alerts_client_factory.ts

```diff
@@ -71,10 +70,9 @@ export class AlertsClientFactory {
         if (!securityPluginSetup) {
           return { apiKeysEnabled: false };
         }
-        const createAPIKeyResult = await securityPluginSetup.authc.createAPIKey(request, {
-          name: `source: alerting, generated uuid: "${uuid.v4()}"`,
-          role_descriptors: {},
-        });
+        const createAPIKeyResult = await securityPluginSetup.authc.grantAPIKeyAsInternalUser(
```

Comment:
I think this is worthy of a comment, since it wasn't immediately obvious to me what was going on.

```suggestion
        // Create an API key using the new grant API - in this case the Kibana system user is creating the
        // API key for the user, instead of having the user create it themselves, which requires api_key
        // privileges 
        const createAPIKeyResult = await securityPluginSetup.authc.grantAPIKeyAsInternalUser(
```